Lib/fontTools/varLib/instancer.py

```diff
@@ -1284,6 +1298,158 @@ def instantiateVariableFont(
     return varfont
 
 
+def axisValuesFromAxisLimits(stat, axisLimits):
+
+    axisRecords = stat.table.DesignAxisRecord.Axis
+    axisValues = stat.table.AxisValueArray.AxisValue
+
+    format4 = [a for a in axisValues if a.Format == 4]
+    nonformat4  = [a for a in axisValues if a not in format4]
+    axisValues = format4 + nonformat4
+
+    axisOrder = {a.AxisOrdering: a.AxisTag for a in axisRecords}
+    pinnedAxes = set(k for k, v in axisLimits.items() if isinstance(v, (float, int)))
+
+    results, seen_axes = [], set()
+    for axisValue in axisValues:
+
+        # Ignore axisValue if it has ELIDABLE_AXIS_VALUE_NAME flag enabled.
+        # Enabling this flag will hide the axisValue in application font menus.
+        # TODO this is too greedy! we need to retain wght axisValues
+        if axisValue.Flags == 2:
+            continue
+
+        if axisValue.Format == 4:
+            axisIndexes = set(r.AxisIndex for r in axisValue.AxisValueRecord)
+            if seen_axes - axisIndexes != seen_axes:
+                continue
+            # TODO fix dup appends 
+            for rec in axisValue.AxisValueRecord:
+                axisTag = axisOrder[rec.AxisIndex]
+                if axisTag not in pinnedAxes:
+                    continue
+                if rec.Value == axisLimits[axisTag]:
+                    seen_axes.add(rec.AxisIndex)
+                    results.append((rec.AxisIndex, axisValue))
+
+        elif axisValue.Format in (1, 3):
+            axisTag = axisOrder[axisValue.AxisIndex]
+            # Add axisValue if it's used to link to another variable font
+            if axisTag not in axisLimits and axisValue.Value == 1.0:
```

Comment:
It seems to be assumed that the axisLimits parameter passed in must have an element for every axis in this font. If so, perhaps clarify that with comment/doc string back at the start of this function defn.

Also, it's not clear why `axisValue.Value == 1.0` is a condition. 

Let's assume the likely case for two distinct but related variable fonts of Roman and Italic counterparts. The italic font should have an axis value table for 'ital' with value 1.0; but then the Roman font should have an axis value table for 'ital' with value 0. Wouldn't you want both to be treated the same (module whether the ELIDABLE flag is set)?

But generalizing, there's nothing saying that only Roman and Italic could be paired in separate variable fonts in that way. As a hypothetical example, one could (for whatever reason) create variable fonts with wght and wdth axes, but have discrete optical size variants within the family. Or, create variable fonts with wdth and opsz axes but have discrete weight variants Regular, Bold, Black.

So, `1.0` is a very specific case being singled out. If there's a particular reason for that, it should be clarified in a comment.

Reply:
Thanks for the in depth comments Peter. In the end, I just used an existing function within this module (instantiateSTAT) since my code was pretty much the same.